src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueActor.scala

```diff
@@ -211,9 +211,22 @@ private[impl] class LaunchQueueActor(
     suspendedLaunchersMessages += actorRef -> deferredMessages
   }
 
+  /**
+    * Fetch an existing actorRef, or create a new actor if the given instance is scheduled.
+    * We don't need an actor ref to handle update for non-scheduled instances.
+    */
+  private def actorRefFor(instance: Instance): Option[ActorRef] = {
+    launchers.get(instance.runSpecId).orElse {
+      if (instance.isScheduled) {
+        logger.info(s"No active taskLauncherActor for scheduled ${instance.instanceId}, will create one.")
+        groupManager.runSpec(instance.runSpecId).map(createAppTaskLauncher)
+      } else None
+    }
+  }
+
   private[this] def receiveInstanceUpdate: Receive = {
     case update: InstanceChange =>
-      launchers.get(update.runSpecId) match {
+      actorRefFor(update.instance) match {
```

Comment:
this is potentially leaking `TaskLauncherActor` instances in following case:
- someone asks to delete an app
- we purge launch queue so stop TaskLauncherActor
- before we adjust goal to Decomissioned, instance update comes
- new `TaskLauncherActor` is created and it will never be stopped

Reply:
How do we handle the case when actor referenced by ActorRef is already terminated but we still have the reference?

Reply:
@meln1k how would we get to that point? This actor watches death of child actors by listening to `Terminated` so I believe all that should be covered